chassembly/src/test/java/org/quiltmc/chasm/lang/TestRendering.java

```diff
@@ -20,10 +34,79 @@ protected void doTest(Path testPath, Path resultPath) throws IOException {
         System.out.println(expected);
 
         Node parsedAgain = Node.parse(expected);
-        StringBuilder actualBuilder = new StringBuilder();
-        parsedAgain.render(renderer, actualBuilder, 1);
-        String actual = actualBuilder.toString();
 
-        Assertions.assertEquals(expected, actual);
+        assertEqual(parsed, parsedAgain);
+    }
+
+    private static void assertEqual(Node expected, Node actual) {
```

Comment:
If you have time spare, it would be great to investigate if implementing `equals` on nodes would cause problems.
If you don't, this is fine and we can worry about it later.